libgnucash/app-utils/gnc-ui-util.c

```diff
@@ -985,7 +992,11 @@ gnc_find_or_create_equity_account (Account *root,
 
     if (account &&
             gnc_commodity_equiv (currency, xaccAccountGetCommodity (account)))
+    {
+        if (equity_type == EQUITY_OPENING_BALANCE && !xaccAccountGetOpeningBalance (account))
```

Comment:
The second test should be unnecessary: If the opening_balance property was already set the account should have been found and returned up at line 970. If it somehow got by there's no harm in setting it again.